File: Data Structures & Algorithms/longest-palindromic-substring/submission-2.go
func longestPalindrome(s string) string {
	ans := ""

	for i := 0; i<len(s);i+=1{
		// odd case
		l, r := i,i
		for l >= 0 && r < len(s) && s[l] == s[r]{
			if len(s[l:r+1]) > len(ans){
				ans = s[l:r+1]
			}
			l -= 1
			r += 1
		}

		// even case
		l,r = i, i+1
		for l >= 0 && r < len(s) && s[l] == s[r]{
			if len(s[l:r+1]) > len(ans){
				ans = s[l:r+1]
			}
			l -= 1
			r += 1
		}
	}
	return ans
}

File: Data Structures & Algorithms/palindromic-substrings/submission-2.py
class Solution:
    def countSubstrings(self, s: str) -> int:

        ans = 0

        for i,char in enumerate(s):
            # each char is a palindrome
            ans += 1

            # odd case, start from i-1, i+1
            l = i-1
            r = i+1

            while l >= 0 and r < len(s):
                if s[l] == s[r]:
                    ans += 1
                    l -= 1
                    r += 1
                    continue
                break

            l = i
            r = i+1
            # even case, start from i, i+1
            while l >= 0 and r < len(s):
                if s[l] == s[r]:
                    ans += 1
                    l -= 1
                    r += 1
                    continue
                break
                
        return ans



            

        